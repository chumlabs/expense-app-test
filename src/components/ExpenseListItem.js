// IMPORTS
// - libraries
import React from 'react';
// import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import moment from 'moment';
import numeral from 'numeral';
// - components
// import { removeExpense } from '../actions/expenseActions';

const ExpenseListItem = ({ id, description, note, amount, createdAt }) => (
  <div>
    <h3>
      <Link to={`/edit/${id}`}>{description}</Link>
    </h3>

    <ul>
      <li>note: {note}</li>
      <li>amount: {numeral(amount / 100).format('$0,0.00')}</li>
      <li>created: {moment(createdAt).format('MMM Do, YYYY')}</li>
    </ul>
  </div>
);

export default ExpenseListItem;

// --------------------------
// // alt version - attempt to make purely presentational - no dispatcha action needed
// const ExpenseListItem = ({
//   id,
//   description,
//   note,
//   amount,
//   createdAt,
//   dispatch,
//   onListItemClick
// }) => (
//   <div>
//     <h3>{description}</h3>
//     <ul>
//       <li>note: {note}</li>
//       <li>amount: {amount}</li>
//       <li>created: {createdAt}</li>
//     </ul>
//     <button onClick={onListItemClick(id)}>Remove Expense</button>
//   </div>
// );

// const mapDispatchToProps = (dispatch, ownProps) => ({
//   onListItemClick: id => {
//     dispatch(removeExpense({ id }));
//   }
// });

// export default connect(null, mapDispatchToProps)(ExpenseListItem);
