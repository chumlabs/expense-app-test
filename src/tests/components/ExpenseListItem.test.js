import React from 'react';
import { shallow } from 'enzyme';
import ExpenseListItem from '../../components/ExpenseListItem';
import expenses from '../fixtures/expenseData'; // mock data

test('should render ExpenseListItem component with expense data', () => {
  const wrapper = shallow(<ExpenseListItem {...expenses[0]} />);

  expect(wrapper).toMatchSnapshot();
});
