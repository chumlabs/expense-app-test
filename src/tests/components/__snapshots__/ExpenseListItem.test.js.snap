// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseListItem component with expense data 1`] = `
<div>
  <h3>
    <Link
      replace={false}
      to="/edit/1"
    >
      item 1 - abc
    </Link>
  </h3>
  <ul>
    <li>
      note: 
      lowest amount, middle date
    </li>
    <li>
      amount: 
      $7.00
    </li>
    <li>
      created: 
      Apr 1st, 2017
    </li>
  </ul>
</div>
`;
